activation/atxdb.go

```diff
@@ -366,111 +357,88 @@ func (db *ActivationDb) ActiveSetSize(epochId types.EpochId) (uint32, error) {
 	return common.BytesToUint32(val), nil
 }
 
+type atxIdAndLayer struct {
+	AtxId   types.AtxId
+	LayerId types.LayerID
+}
+
 // addAtxToEpoch adds atx to epoch epochId
 // this function is not thread safe and needs to be called under a global lock
-func (db *ActivationDb) addAtxToEpoch(epochId types.EpochId, atx types.AtxId) error {
-	ids, err := db.atxs.Get(epochId.ToBytes())
-	var atxs []types.AtxId
+func (db *ActivationDb) updateTopAtxIfNeeded(atx *types.ActivationTx) error {
+	currentTopAtx, err := db.getTopAtx()
+	if err != nil && err != database.ErrNotFound {
+		return fmt.Errorf("failed to get current ATX: %v", err)
+	}
+	if err == nil && currentTopAtx.LayerId >= atx.PubLayerIdx {
+		return nil
+	}
+
+	newTopAtx := atxIdAndLayer{
+		AtxId:   atx.Id(),
+		LayerId: atx.PubLayerIdx,
+	}
+	topAtxBytes, err := types.InterfaceToBytes(&newTopAtx)
 	if err != nil {
-		//epoch doesnt exist, need to insert new layer
-		ids = []byte{}
-		atxs = make([]types.AtxId, 0, 0)
-	} else {
-		atxs, err = decodeAtxIds(ids)
-		if err != nil {
-			return errors.New("could not get all atxs from database ")
-		}
+		return fmt.Errorf("failed to marshal top ATX: %v", err)
 	}
-	atxs = append(atxs, atx)
-	w, err := encodeAtxIds(atxs)
+
+	err = db.atxs.Put([]byte(topAtxKey), topAtxBytes)
```

Comment:
Keep `topAtxKey` as []byte instead of casting every time you use it

Reply:
I chose a string because it's immutable and can be a constant (unlike a byte array).

I could store it in a global variable instead and save me the casting, but I don't think it's that much better. If you insist, I'll do it.